frame/gilt/src/tests.rs

```diff
@@ -0,0 +1,499 @@
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 80, 1));
+		Gilt::enlarge(40, 2);
+
+		// Takes 2/2, then stopped because it reaches its max amount
+		assert_eq!(Queues::<Test>::get(1), vec![ GiltBid { amount: 40, who: 1 } ]);
+		assert_eq!(QueueTotals::<Test>::get(), vec![(1, 40), (0, 0), (0, 0)]);
+
+		assert_eq!(Active::<Test>::get(0).unwrap(), ActiveGilt {
+			proportion: Perquintill::from_percent(10),
+			amount: 40,
+			who: 1,
+			expiry: 4,
+		});
+		assert_eq!(ActiveTotal::<Test>::get(), ActiveGiltsTotal {
+			frozen: 40,
+			proportion: Perquintill::from_percent(10),
+			index: 1,
+			target: Perquintill::zero(),
+		});
+	});
+}
+
+#[test]
+fn basic_thaw_works() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 40, 1));
+		Gilt::enlarge(40, 1);
+		run_to_block(3);
+		assert_noop!(Gilt::thaw(Origin::signed(1), 0), Error::<Test>::NotExpired);
+		run_to_block(4);
+		assert_noop!(Gilt::thaw(Origin::signed(1), 1), Error::<Test>::Unknown);
+		assert_noop!(Gilt::thaw(Origin::signed(2), 0), Error::<Test>::NotOwner);
+		assert_ok!(Gilt::thaw(Origin::signed(1), 0));
+
+		assert_eq!(ActiveTotal::<Test>::get(), ActiveGiltsTotal {
+			frozen: 0,
+			proportion: Perquintill::zero(),
+			index: 1,
+			target: Perquintill::zero(),
+		});
+		assert_eq!(Active::<Test>::get(0), None);
+		assert_eq!(Balances::free_balance(1), 100);
+		assert_eq!(Balances::reserved_balance(1), 0);
+	});
+}
+
+#[test]
+fn thaw_when_issuance_higher_works() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 100, 1));
+		Gilt::enlarge(100, 1);
+
+		// Everybody else's balances goes up by 50%
+		Balances::make_free_balance_be(&2, 150);
+		Balances::make_free_balance_be(&3, 150);
+		Balances::make_free_balance_be(&4, 150);
+
+		run_to_block(4);
+		assert_ok!(Gilt::thaw(Origin::signed(1), 0));
+
+		assert_eq!(Balances::free_balance(1), 150);
+		assert_eq!(Balances::reserved_balance(1), 0);
+	});
+}
+
+#[test]
+fn thaw_when_issuance_lower_works() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 100, 1));
+		Gilt::enlarge(100, 1);
+
+		// Everybody else's balances goes down by 25%
+		Balances::make_free_balance_be(&2, 75);
+		Balances::make_free_balance_be(&3, 75);
+		Balances::make_free_balance_be(&4, 75);
+
+		run_to_block(4);
+		assert_ok!(Gilt::thaw(Origin::signed(1), 0));
+
+		assert_eq!(Balances::free_balance(1), 75);
+		assert_eq!(Balances::reserved_balance(1), 0);
+	});
+}
+
+#[test]
+fn multiple_thaws_works() {
+	new_test_ext().execute_with(|| {
+		run_to_block(1);
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 40, 1));
+		assert_ok!(Gilt::place_bid(Origin::signed(1), 60, 1));
+		assert_ok!(Gilt::place_bid(Origin::signed(2), 50, 1));
+		Gilt::enlarge(200, 3);
+
+		// Double everyone's free balances.
+		Balances::make_free_balance_be(&2, 100);
```

Comment:
is this correct? 2 is not everyone else, nor is the 100 double its original balance.

Reply:
The comment refers to the follow three lines, which is everyone else and does double their balances.